recipe-java-runtime/src/main/java/ca/derekcormier/recipe/BaseIngredient.java

```diff
@@ -46,6 +47,32 @@ protected void setProperty(String key, Object value) {
         return (T)properties.get(key);
     }
 
+    protected <T> T getProperty(Class<T> clazz, String key) {
+        if (hasProperty(key)) {
+            return (T)getProperty(key);
+        }
+        else if (clazz == int.class) {
+            return (T)new Integer(0);
+        }
+        else if (clazz == float.class) {
+            return (T)new Float(0.0f);
+        }
+        else if (clazz == boolean.class) {
+            return (T)new Boolean(false);
+        }
+        else if (clazz == String.class) {
```

Comment:
This block can probably be dropped and just let it fall-through to the default `return null`?